core/core_integrations/slack.js

```diff
@@ -17,77 +17,77 @@ inTransaction = {},
 waitingForFirstTransaction = {},
 clearedMessageQueue = false,
 clearedTimeouts = false,
-tempTeamData = {},
+teamData = {},
 
-sendMessage = function(message, thread) {
-	var teamInfo = getChannelIdAndTeamId(thread);
+sendMessageWebAPI = function (teamId) {
+	// Initialise for team
+	if (inTransaction[teamId] === null) {
+		inTransaction[teamId] = false;
+    }
 
-	if (teamInfo.token !== null) {
-		var body = {
-			"token": teamInfo.token,
-			"channel": teamInfo.channel_id,
-			"username": exports.config.name,
-			"link_names": 1,
-			"text": message,
-			"unfurl_links": true,
-			"icon_url": exports.config.icon
-		};
-		addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/chat.postMessage');
-	}
-	else {
-		console.debug('slack-> No slack team found!!!');
+	if (messageQueue[teamId].length >= 1 && !inTransaction[teamId]) {
+		inTransaction[teamId] = true;
+		var message = messageQueue[teamId].shift();
+		request({
+				uri: message.uri,
+				method: 'GET',
+				qs: message.body
+			},
+			function (error, response, body) {
+				body = JSON.parse(body);
+				if (error) {
+					console.debug('slack-> error: ' + error);
+					message.callback(true, null);
+				} else if (!body.ok) {
+					console.debug('slack-> Failed to send message, error: ' + body.error);
+					message.callback(true, null);
+				} else if (message.callback) {
+					message.callback(false, body);
+				}
+				inTransaction[teamId] = false;
+				// Wait until message is sent before trying to send another message.
+				sendMessageWebAPI(teamId);
+			}
+		);
 	}
 },
 
-addMessageToQueue = (teamId, body, uri, callback) => {
+addMessageToQueue = function(teamId, body, uri, callback) {
 	waitingForFirstTransaction[teamId] = false;
 	if (!messageQueue[teamId]) {
 		messageQueue[teamId] = [];
 	}
-	messageQueue[teamId].push({"body": body, "uri": uri, "callback": callback});
+	messageQueue[teamId].push({body: body, uri: uri, callback: callback});
 	sendMessageWebAPI(teamId);
 },
 
-sendMessageWebAPI = function (teamId) {
-	// Initialise for team
-	if (inTransaction[teamId] === null || inTransaction[teamId] === undefined) {
-		inTransaction[teamId] = false;
-    }
+sendMessage = function(message, thread) {
+	var teamInfo = getChannelIdAndTeamId(thread);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["getChannelIdAndTeamId" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2549394133/issues/source?bid=3282987&fileBranchId=3288593#l65)